src/hooks/useClientScriptLoad.ts

```diff
@@ -0,0 +1,82 @@
+import { useState, useMemo, useEffect } from "react"
+import { isNostoLoaded } from "../components/helpers"
+import type { NostoClient } from "../types"
+import type { NostoProviderProps } from "../components/NostoProvider"
+
+type NostoScriptProps = Pick<NostoProviderProps, "account" | "host" | "shopifyMarkets">
+
+export function useClientScriptLoad(props: NostoScriptProps) {
```

Comment:
rename this to `useLoadClientScript` or `useClientScriptLoader`

Reply:
useLoadClientScript would be my preference